### PR TITLE
Replaced append(Period) with add_interval(kwargs)

### PR DESCRIPTION
**Pull request by BSchilperoort**
This PR will tackle Peter's comments about the Period classes:

- [x] Perhaps we should hide the Period class from the public API. Why not `calendar.add_target_period(...)` and `add_precursor_period`? I found myself doing `calendar.append(PrecursorPeriod("1d"))`, which I think is a sign of suboptimal API
- [x] Relatedly, should it be `add_target`, `add_target_period`, `add_target_interval`, `add_period(target=True)`, `add_interval(target=True)`, or `add_block(target=True)`? What's the difference between a precursor and targetperiod?
- [x] ~I would love to be able to add a period by specifying a length and end day, e.g. if I want to issue a forecast by the end of may, I want to start appending precursors from 31 may backwards.~ See #137 

**Changes:**
 - Intervals are appended using `calendar.add_interval("target", "10d", gap="2d')`
     - TargetPeriod and PrecursorPeriod classes are still used in the backend.
 - Support for non-day lengths and gaps is added, required a rewrite of `skip_nyears`
   - E.g., `calendar.add_interval("target", "2M")`
 - Tests and notebook have been updated
 - `mypy` version is pinned, see #134.

### CODE REVIEW

File: tests/test_custom_calendar.py
Comment (geek-yang) [inline tests/test_custom_calendar.py:139]: Would be nice to have one more test for the `allow_overlap` case.

**Review by geek-yang — COMMENTED** [review]
Looks really nice to me. I checked the codes and the notebook. The tasks are clearly defined and the implementations are quite clean. Only got one minor comment, which concerns the test. I would wait for the feedback from @Peter9192, before this PR can be merged. Nice work!

**Comment by Peter9192** [comment]
closes #134

**Comment by Peter9192** [comment]
> * Intervals are appended using `calendar.append("target", "10d", gap="2d')`

`calendar.add_interval`, right?

**Review by Peter9192 — APPROVED** [review]
Nice!

**Comment by BSchilperoort** [comment]
> > * Intervals are appended using `calendar.append("target", "10d", gap="2d')`
> 
> `calendar.add_interval`, right?

Yes! I mistyped that.

**Pull request merged**
